NSS-Day-Cohort-42/rare-media-moguls
# Filter Posts by Author from User Profile View

#### Changes Made
1. Updated 'UserPostList.js' to include conditional logic in a useEffect to get posts by author or by user (to not upset the My Posts functionality)
2. Updated 'UserProfile.js' with the current link and passed props through link (check it out, it is nifty)
3. Added conditional logic to 'Post.js' to only see edit and delete buttons if current user is viewing their own posts. 
4. Added a different view to Post Routes!

#### Steps to Review
1. Checkout this branch locally and run the application.
```
git fetch --all
git checkout sm-postsbyauthor
```
2. Run the application.
```
npm start
```
3. Initialize virtual environment, and run the server.
```
pipenv shell
python manage.py server
```
4. Test app functionality.
> Navigate to a user's profile by clicking on the author's name on post. Click on the link that says how many posts that author currently has published. 
>That link should take you to a page (looks just like the my posts link) that allows you to view all of their posts.
>Check that if the user is viewing posts other than their own, that the edit and delete buttons are not populating!

5. View code file.
> Confirm file modifications are present as indicated above.
> Confirm no unused code or extraneous comments exist.
